### PR TITLE
Set a max length of 100 for most inputs

### PR DESCRIPTION
**Pull request by jasonhenriquez**
Excluded are playlist descriptions, which presumably deserve to have some leeway, and the password input, to prevent the edge case of a user being locked out if they had a longer one set (and also why not for this one).

# Set a max length of 100 for most inputs

## Pull Request Type

- [x] Feature Implementation

## Related issue
https://github.com/FreeTubeApp/FreeTube/pull/5150#issuecomment-2124043618

## Description
There's just no need for that much information in a custom label, because beyond 100 (and before, in many cases), we're already limiting the visible text. 

Excluded are playlist descriptions, which presumably deserve to have some leeway, and the password input, to prevent the edge case of a user being locked out if they had a longer one set (and also why not for this one)

**Note:** Due to the value of 100 chosen, [this code](https://github.com/FreeTubeApp/FreeTube/pull/4992/files) is currently not in use. We can 1) keep this code here in case that number ever changes, or 2) remove it and just remember to revert a commit. I don't have a preference.

## Testing 
- Test that inputs are limited from continued text at 100
- (Optional) Test edge case behavior when a playlist title is already >100 characters and the title is edited. Expected outcome: you can't add new characters, but you can remove them, and save at any length.

## Desktop

- **OS:** OpenSUSE
- **OS Version:** TW

## Additional context
The User Playlist desktop list view breaks for long enough titles and descriptions. We probably need code truncating those after x lines.

### CODE REVIEW

**Comment by absidue** [comment]
Please make the max length opt-in, that means the default value is null and then you pass in the limit in the few places it is needed, making it opt-out is going to cause more problems than it will solve. For example you currently set a 100 character limit on the search bar, with the reasoning that it makes the search query limit code obsolete, however that will break the second use of the search bar: handling URLs. The URLs can easily go over 100 characters, especially if they come from a tor Invidious instance, see this comment for more information: https://github.com/FreeTubeApp/FreeTube/pull/4288#issuecomment-1800490629

**Comment by jasonhenriquez** [comment]
Thanks for this context @absidue, I wasn't aware of that. Will update.

**Comment by jasonhenriquez** [comment]
Updated now to remove the constraints from places where it would restrict valid inputs. Kept the default value the same so we don't have to have a magic number, or a constant we'd have to import into each file.

**Comment by PikachuEXE** [comment]
> Please make the max length opt-in, that means the default value is null and then you pass in the limit in the few places it is needed, making it opt-out is going to cause more problems than it will solve.

I don't think this is solved
And also now I have no idea what to test (coz the diff doesn't show the list of inputs

**Comment by jasonhenriquez** [comment]
> I don't think this is solved

See answer:
> Kept the default value the same so we don't have to have a magic number, or a constant we'd have to import into each file.

>  And also now I have no idea what to test (coz the diff doesn't show the list of inputs

Thanks, just updated the OP with the list of non-affected inputs.

**Review by PikachuEXE — CHANGES_REQUESTED** [review]
- Playlist Name should be 255 at least (or unlimited)
- Playlist Description should be 255 at least (or unlimited)
- Proxy Port Number should be 5

Also in https://github.com/FreeTubeApp/FreeTube/pull/5150#issuecomment-2124043618 I simply means limiting the displayed value not input value

File: src/renderer/components/ft-input/ft-input.js [hunk 1]
```diff
@@ -23,6 +23,10 @@ export default defineComponent({
       type: String,
       default: null
     },
+    maxlength: {
+      type: Number,
+      default: 100
```
[the remark this answers is not in the document]
Comment (jasonhenriquez) [reply src/renderer/components/ft-input/ft-input.js:28]: 100 is just a number I chose that's arbitrarily high but minimizes UI breaking / misleading the user as to how much of their input will realistically be displayed when applicable. If anything, it's probably a bit too high, but I think it's fine. I don't think there's any reason to have different individualized values for each input.

[the remark this answers is not in the document]
Comment (PikachuEXE) [reply src/renderer/components/ft-input/ft-input.js:28]: Places with 100 char limit: (13)
- src/renderer/views/UserPlaylists/UserPlaylists.vue - searchBar
- src/renderer/views/SubscribedChannels/SubscribedChannels.vue - searchBar
- src/renderer/views/History/History.vue - searchBar
- src/renderer/views/Channel/Channel.vue - searchBar
- src/renderer/components/proxy-settings/proxy-settings.vue - proxyPort
- src/renderer/components/playlist-info/playlist-info.vue - Playlist Name
- src/renderer/components/playlist-info/playlist-info.vue - Playlist Description
- src/renderer/components/playlist-info/playlist-info.vue - Search for Videos
- src/renderer/components/ft-profile-edit/ft-profile-edit.vue - Profile Name
- src/renderer/components/ft-playlist-add-video-prompt/ft-playlist-add-video-prompt.vue - searchBar
- src/renderer/components/ft-create-playlist-prompt/ft-create-playlist-prompt.vue - Playlist Name
- src/renderer/components/distraction-settings/distraction-settings.vue - Hide Channels
- src/renderer/components/distraction-settings/distraction-settings.vue - Hide Videos and Playlists Containing Text

Places with unlimited length: 12 (`:maxlength="null"`)


Youtube playlist name now has 150 char limit, description has 5000 char limit:
image

image


Channel name now has 100 char limit:
https://www.reddit.com/r/JackSucksAtLife/comments/17e8h08/since_when_was_the_youtube_channel_name_limit/
image


This means these places should have 255 chars limit due to Playlist Name + Channel Name = 250 + 5 char for whatever purpose like delimiter:
- src/renderer/views/UserPlaylists/UserPlaylists.vue - searchBar
- src/renderer/components/playlist-info/playlist-info.vue - Playlist Name
- src/renderer/components/playlist-info/playlist-info.vue - Playlist Description (not saying it should match Youtube's 5000 char limit but the limit should be at least the same as playlist name or larger)
- src/renderer/components/ft-playlist-add-video-prompt/ft-playlist-add-video-prompt.vue - searchBar
- src/renderer/components/ft-create-playlist-prompt/ft-create-playlist-prompt.vue - Playlist Name

Other places:
- src/renderer/components/distraction-settings/distraction-settings.vue - Hide Videos and Playlists Containing Text (150 due to playlist anme)
- src/renderer/components/proxy-settings/proxy-settings.vue - proxyPort (max value 65535 = 5 chars)


And now there are 6 places left with 100 char limit while 12 places with no limit and I have no idea why there is a default value of 100

[the remark this answers is not in the document]
Comment (jasonhenriquez) [reply src/renderer/components/ft-input/ft-input.js:28]: I'm tired and having a bit of trouble following. What's the rationale for most of the suggested changes beyond conforming to YT? I did mean to remove the limit on the playlist description, but I must have accidentally removed it, so I can do that. My general idea is that we can use one arbitrary not-too-large value for most custom name-like fields and unbounded for ones where we don't have a reason to care.

[the remark this answers is not in the document]
Comment (jasonhenriquez) [reply src/renderer/components/ft-input/ft-input.js:28]: Re-reading this, I come off as unnecessarily callous. My main point is that I'm willing to spend at most ten minutes on further changes for this PR, because I do think we (and developers in the FOSS community as a whole) have a tendency to bikeshed about obscure minutia in a way that can be quite counterproductive. I worry that if I were to implement your suggested changes in full, I'm going to get a review from another person saying to do it all another way, and the process just gives me the heebie jeebies sometimes. So if you are confident that you have a set of changes that you think I can do pretty quickly that are also unlikely to be opposed by any other developer on our team, I'm game. Just give me the sign.

[the remark this answers is not in the document]
Comment (PikachuEXE) [reply src/renderer/components/ft-input/ft-input.js:28]: Just for changes to be made: change char limit to 255 in following places:
- src/renderer/views/UserPlaylists/UserPlaylists.vue - searchBar
- src/renderer/components/playlist-info/playlist-info.vue - Playlist Name
- src/renderer/components/ft-playlist-add-video-prompt/ft-playlist-add-video-prompt.vue - searchBar
- src/renderer/components/ft-create-playlist-prompt/ft-create-playlist-prompt.vue - Playlist Name

Reasons: 
- Remote playlist can have 150 chars
- Channel name can have 100 chars (and some people might want to add them)
- Existing display logic truncate string to 255 chars

**Review by PikachuEXE — APPROVED** [review]

**Comment by absidue** [comment]
I still think it would be better to get rid of the implicit limit of 100 and instead set explicit limits where they are actually required. Your argument was that we would need a magic constant imported into multiple files, but I would argue that we don't need a constant at all, just write the number where it is needed. Secondly that the current situation is worse, because it's still a magic number, but you don't see it at the call/usage site, so you won't even know it exists unless you stumble upon it in testing or remember to look at the `ft-input` component. It also means we won't have to override it with `null` in so many places anymore.

File: src/renderer/components/download-settings/download-settings.vue [hunk 2]
```diff
@@ -34,6 +34,7 @@
             :show-action-button="false"
             :show-label="false"
             :disabled="true"
+            :maxlength="100"
```
Comment (absidue) [inline src/renderer/components/download-settings/download-settings.vue:37]: As this displays the users chosen downloads folder, the path could much longer than 100 characters.
```suggestion
```

File: src/renderer/components/player-settings/player-settings.vue [hunk 3]
```diff
@@ -247,6 +247,7 @@
           :spellcheck="false"
           :show-action-button="false"
           :show-label="false"
+
```
Comment (absidue) [inline src/renderer/components/player-settings/player-settings.vue:250]: Extra whitespace.
```suggestion
```

File: src/renderer/components/proxy-settings/proxy-settings.vue [hunk 4]
```diff
@@ -29,6 +29,7 @@
           :show-action-button="false"
           :show-label="true"
           :value="proxyHostname"
+
```
Comment (absidue) [inline src/renderer/components/proxy-settings/proxy-settings.vue:32]: Ditto
```suggestion
```

File: src/renderer/components/sponsor-block-settings/sponsor-block-settings.vue [hunk 5]
```diff
@@ -40,6 +40,7 @@
           :show-action-button="false"
           :show-label="true"
           :value="sponsorBlockUrl"
+
```
Comment (absidue) [inline src/renderer/components/sponsor-block-settings/sponsor-block-settings.vue:43]: Ditto
```suggestion
```

File: src/renderer/components/sponsor-block-settings/sponsor-block-settings.vue [hunk 6]
```diff
@@ -52,6 +53,7 @@
           :show-action-button="false"
           :show-label="true"
           :value="deArrowThumbnailGeneratorUrl"
+
```
Comment (absidue) [inline src/renderer/components/sponsor-block-settings/sponsor-block-settings.vue:56]: Ditto
```suggestion
```

File: src/renderer/components/top-nav/top-nav.vue [hunk 7]
```diff
@@ -87,6 +87,7 @@
           :data-list="searchSuggestionsDataList"
           :spellcheck="false"
           :show-clear-text-button="true"
+
```
Comment (absidue) [inline src/renderer/components/top-nav/top-nav.vue:90]: Ditto
```suggestion
```

**Comment by absidue** [comment]
Other than that it looks good now, thank you for doing the changes.

**Review by absidue — APPROVED** [review]

**Review by PikachuEXE — APPROVED** [review]

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]
